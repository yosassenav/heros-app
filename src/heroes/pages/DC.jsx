export const DC = () => {
    return <h1>DC</h1>
}