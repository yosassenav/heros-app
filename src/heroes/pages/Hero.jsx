import { Navigate, useParams } from "react-router-dom"
import { getHeroById } from "../helpers";

export const Hero = () => {

    const { id } = useParams();

    const hero = getHeroById( id );

    if ( !hero ) {
        return <Navigate to={"/marvel"}/>
    }

    
    return <h1>{hero.superhero}</h1>
}