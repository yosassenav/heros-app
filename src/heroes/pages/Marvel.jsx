export const Marvel = () => {
    return <h1>Marvel</h1>
}