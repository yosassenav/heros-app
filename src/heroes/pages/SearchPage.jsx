import { useNavigate } from "react-router-dom";
import { useForm } from "../../hooks/useForm"
import { HeroCard } from "../components"

export const SearchPage = () => {

    const navigate = useNavigate();

    const {searchText, onInputChange} = useForm({
        searchText: '',
    });

    const onSearchSubmit = (event) => {
        event.preventDefault();

        if(searchText.trim().length <= 1) return;
        navigate(`?q=${ searchText }`);


        console.log({searchText})
    }

    return (
    <>
        <h1>Search</h1>
        <hr/>

        <div className="row">
            <div className="col-5">
                <form onSubmit={ onSearchSubmit }>
                    <input type="text"
                    className="form-control"
                    placeholder="Search a hero"
                    name="searchText"
                    autoComplete="off"
                    value={ searchText }
                    onChange={ onInputChange }
                    />
                </form>
                <button className="btn btn-outline-primary mt-2">
                    Search
                </button>
            </div>
            <div className="col-7">
                <h4>Results</h4>
                <hr/>

                <div className="alert alert-primary">
                    Search a hero
                </div>
                <div className="alert alert-danger">
                    Hero not found!
                </div>

                <HeroCard/>

            </div>
        </div>
        
        
    </>
)
}