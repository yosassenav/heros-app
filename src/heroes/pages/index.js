export * from "./DC"
export * from "./Marvel"