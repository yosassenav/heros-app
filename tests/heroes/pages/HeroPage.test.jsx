/**
 * @jest-environment jsdom
 */

import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import { HeroPage } from "../../../src/heroes/pages/HeroPage"
import { MemoryRouter } from "react-router-dom"

// mock of useNavigate hook from react testing library
const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom',()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: ()=>mockedUseNavigate
}))

describe('Pruebas en <HeroPage/>',()=>{

    beforeAll(()=>{
        jest.clearAllMocks();
    })

    test('debe mostrarse correctamente con los valores por defecto',()=>{
        const {container} = render(
            <MemoryRouter>
                <HeroPage/>
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot();
    })

    test('debe regresar a MarvelPage al hacer click en el boton Back',()=>{
        render(
            <MemoryRouter initialEntries={['/HeroPage']}>
                <HeroPage/>
            </MemoryRouter>
        )

        const btn = screen.getByRole('button')
        fireEvent.click(btn);
        expect(mockedUseNavigate).toHaveBeenCalledTimes(1);


    })
    
})