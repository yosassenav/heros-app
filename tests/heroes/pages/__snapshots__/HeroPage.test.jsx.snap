// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroPage/> debe mostrarse correctamente con los valores por defecto 1`] = `<div />`;
